### PR TITLE
CI: migrate pipeline into GH Actions Apache

### PR DESCRIPTION
**Pull request by brushknight**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

File: build/package/windows/nri-386-installer/Product.wxs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+    <?define Version = "$(var.IntegrationVersion)" ?>
+    <?define ProductId = "*" ?>
+    <?define UpgradeCode = "{1a921296-1d82-4014-83ad-9d0c2f9c9bef}" ?>
```
Comment (gsanchezgavier) [inline build/package/windows/nri-386-installer/Product.wxs:5]: upgrade code repeated with other integrations pr for the ci

**Review by gsanchezgavier — CHANGES_REQUESTED** [review]

**Comment by paologallinaharbur** [comment]
Going to close this in favor of https://github.com/newrelic/nri-apache/pull/52

**Comment by paologallinaharbur** [comment]
This is ready to be reviewed

**Review by roobre — APPROVED** [review]
lgtm!

**Review by arvdias — APPROVED** [review]

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+on:
+  release:
+    types:
+      - prereleased
+    tags:
+      - 'v*'
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply .github/workflows/prerelease.yml:11]: I believe it is not, but I am not 100% sure

[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply .github/workflows/prerelease.yml:11]: https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret

The environment variable is not created automatically

**Pull request merged**